src/main/java/org/teraslogy/simpleliquids/block/entity/LiquidEntitySystem.java

```diff
@@ -143,4 +148,44 @@ public void update(float delta) {
             timeSinceLastUpdate -= 0.3f;
         }
     }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL)
+    public void onChunkGenerated(OnChunkGenerated chunkGenerated, EntityRef entity) {
+        chunkWorker(chunkGenerated.getChunkPos(), entity);
+    }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL)
+    public void onChunkLoaded(OnChunkLoaded chunkLoaded, EntityRef entity) {
+        chunkWorker(chunkLoaded.getChunkPos(), entity);
+    }
+
+    public void chunkWorker(Vector3i chunkPos, EntityRef entity) {
+        Vector3i worldPos = new Vector3i(chunkPos);
+        worldPos.mul(ChunkConstants.SIZE_X, ChunkConstants.SIZE_Y, ChunkConstants.SIZE_Z);
+        Vector3i blockPos = new Vector3i();
+        Vector3i testPos = new Vector3i();
+        // scan the chunk, looking for liquid
+        for (int y = ChunkConstants.SIZE_Y - 1; y >= 0; y--) {
+            for (int z = 0; z < ChunkConstants.SIZE_Z; z++) {
+                for (int x = 0; x < ChunkConstants.SIZE_X; x++) {
+                    blockPos.set(x + worldPos.x, y + worldPos.y, z + worldPos.z);
+                    if (worldProvider.getBlock(blockPos).isLiquid()) {
+                        // scan the neighboring blocks
+                        for (Side side : Side.horizontalSides()) {
+                            testPos.set(blockPos.x, blockPos.y, blockPos.z);
+                            testPos.add(side.getVector3i());
+                            // we only do this if we have air next to our liquid
+                            if (worldProvider.getBlock(testPos) == air) {
+                                blockUpdate(new OnChangedBlock(blockPos, water, water), worldProvider.getBlock(blockPos).getEntity());
+                                break; // GET TO THE CHOPPAH!
+                            }
+                        }
+                        // ___.___
+                        //  c00D`=--/
```

Comment:
The `break;` has the comment `GET TO THE CHOPPAH!`  So I added a helicopter to indicate where the instruction pointer lands upon executing the `break;`